test/kv.go

```diff
@@ -547,11 +558,12 @@ func snapshot(kv kvdb.Kvdb, t *testing.T) {
 	kvPairs, err := snap.Enumerate(prefix)
 	assert.NoError(t, err, "Unexpected error on Enumerate")
 
-	assert.Equal(t, len(kvPairs), count,
-		"Expecting %d keys under %s got: %d, kv: %v",
-		count, prefix, len(kvPairs), kvPairs)
-
+	processedKeys := 0
 	for i := range kvPairs {
+		if strings.Contains(kvPairs[i].Key, keyd) {
```

Comment:
this needs to change.
if deleted key is found - then snapVersion must be lower than delete version.
if deleted key is not found - snapVersion must be higher than delete version.

Reply:
In both the cases the delete version will be less than snapVersion as delete does not really update the index of the key, but returns the one which was during the last Put which is always going to be less than the snapVersion as all the delete keys have been put before snap is taken.

Reply:
that seems like a bug. every modification (get/put/delete) of kvdb should increase the kvdb index by 1 ?

Reply:
for test purposes, CompareAndDelete should work.

Reply:
etcd v3 for ex. does not return the key which was deleted, hence we do not know the index. The way we handle it in our API is we do a Get first to get the kvp and then perform the Delete. May be we can change this API if we want the latest index as well.

Also not sure if I understand your CompareAndDelete comment.

Reply:
the test should be like this:
1. if snap.Enumerate returns deleted key, then go function ran update after snapshot, so delete version of key should be higher than snap version
2. if snap.Enumerate does not find deleted key, then delete version of key must be lower than snap version
if Delete() is not returning the delete version, you can use CompareAndDelete which probably returns the delete version.